compiler-core/src/language_server/tests/action.rs

```diff
@@ -78,18 +82,22 @@ fn apply_code_edit(
 ) -> String {
     let mut result = src.to_string();
     let line_numbers = LineNumbers::new(src);
-    let mut offset = 0;
+    let mut offset: i32 = 0;
     for (change_url, change) in changes {
         if url != change_url {
             panic!("Unknown url {}", change_url)
         }
+        let mut change = change.to_owned();
+        change.sort_by_key(|f| f.range.start);
         for edit in change {
-            let start =
-                line_numbers.byte_index(edit.range.start.line, edit.range.start.character) - offset;
-            let end =
-                line_numbers.byte_index(edit.range.end.line, edit.range.end.character) - offset;
+            let start = line_numbers.byte_index(edit.range.start.line, edit.range.start.character)
+                as i32
+                + offset;
+            let end = line_numbers.byte_index(edit.range.end.line, edit.range.end.character) as i32
+                + offset;
             let range = (start as usize)..(end as usize);
-            offset += end - start;
+            offset -= end - start;
+            offset += edit.new_text.len() as i32;
```

Comment:
What are these changes for? I couldn't figure it out.

Reply:
I have added some comments to the code to help explain what its attempting to do.

The reason these changes were needed is that the previous code only worked on text edits that removed text. The offset was used to update to locations in subsequent edits in the same batch. As the annotate types action adds text, a solely negative offset no longer worked.

Reply:
Fab, thank you